CPP_STYLE_GUIDE.md

```diff
@@ -166,18 +167,34 @@ class FancyContainer {
  ...
 }
 ```
-## Memory Management
 
-### Memory allocation
+## Semantics
+
+The Node.js C & C++ code strives to be consistent in it's use of language
```

Comment:
It's a bit odd to put this section here...can we move it to the beginning of the document? We do inherit some formatting styles from Google's C++ style anyway.

Reply:
typo: its

Reply:
Ack x 2.
I'll try to reformat this.